libs/filamentapp/src/NativeWindowHelperCocoa.mm

```diff
@@ -42,6 +42,12 @@
     // full-screen, we can skip the macOS compositor if the size matches the display size.
     metalLayer.drawableSize = [view convertSizeToBacking:view.bounds.size];
 
+    // In its implementation of vkGetPhysicalDeviceSurfaceCapabilitiesKHR, MoltenVK takes into
+    // consideration both the size (in points) of the bounds, and the contentsScale of the
+    // CAMetalLayer from which the Vulkan surface was created.
+    // See also https://github.com/KhronosGroup/MoltenVK/issues/428
+    metalLayer.contentsScale = view.window.backingScaleFactor;
```

Comment:
@bejado PTAL at this codeline and the one in resize, since these code paths are active with our Metal backend too.  AFAICT it has no effect with our Metal backend.

Reply:
Well, this would affect the Metal backend, but only on desktop. This looks fine to me, if you run with the Metal backend does scaling work correctly still?

Reply:
@bejado Yep it looks fine with `-a metal` on both my monitors.